doc/src/dev/coding_style.rst

```diff
@@ -0,0 +1,217 @@
+
+Coding Style
+=============
+
+Naming Conventions
+~~~~~~~~~~~~~
+
+Consistent naming makes code easier to read, search, and maintain.
+The following rules help enforce a uniform naming style throughout the codebase:
+
+General Rules
+-------------
+
+- Use descriptive and unambiguous names.
+- Avoid abbreviations unless they are widely understood (e.g., `id`, `ctx`, `cfg`).
+- Prefer clarity over brevity.
+
+Specific Conventions
+--------------------
+
+- **Class names** use `PascalCase`.
+
+  .. code-block:: cpp
+
+      class DecompNetlistRouter {
+      };
+
+- **Variable and function names** use `snake_case`.
+
+  .. code-block:: cpp
+
+      int net_count;
+      void estimate_wirelength();
+
+- **Private member variables and private methods** should end with an underscore `_`.
+
+  .. code-block:: cpp
+
+      class Example {
```

Comment:
Note: this convention is not followed very consistently today but we should move to it.

Reply:
Should we say that?

Reply:
Or maybe we just say that at the top of the style guide.